cmake/autodetect_executors.cmake

```diff
@@ -14,7 +14,10 @@ if (NOT DEFINED GINKGO_BUILD_OMP)
     endif()
 endif()
 
-if (NOT DEFINED GINKGO_BUILD_MPI)
+if(NOT DEFINED GINKGO_BUILD_MPI)
+    set(GINKGO_MPI_ASSUME_NO_BUILTIN_MPI ON CACHE BOOL "Disables using MPI wrapper compiler directly.")
+    mark_as_advanced(GIKNGO_MPI_ASSUME_NO_BUILTIN_MPI)
```

Comment:
Do we need to make this controllable at all? It seems to be relevant only in very limited circumstances, where either `mpiexec` is missing from the environment, or the user has a MPI compiler wrapper loaded that is not the one specified by `CXX_COMPILER`

Reply:
TBH, I'm not 100% sure how cmake finds MPI with this option on. Perhaps there are some weird cases, where it doesn't use the same MPI as the MPI wrapper would have. So I added the option to turn it off. I also marked it as advanced to indicate that users generally need not to worry about it.